### PR TITLE
v.cluster: Add tests

### PR DESCRIPTION
**Pull request by Kiraschild**
This PR introduces comprehensive testing for the v.cluster module after refining my initial approach. I've implemented robust validation for DBSCAN clustering, including parameter sensitivity tests for distance and min_points, 2D/3D handling with the -2 flag, and edge cases like minimal clusters and noise points. The tests now feature proper cleanup routines and resolve previous CI issues (like CRLF line endings)

### CODE REVIEW

**Review by echoix — APPROVED** [review]

File: vector/v.cluster/testsuite/cluster_test.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        for line in ascii_output.splitlines():
+            if line.strip():
+                parts = line.split(",")
+                if len(parts) >= 3:
+                    cluster_id = int(parts[2])
+                    if cluster_id == 0:
+                        noise_points.append(cluster_id)
+
+        return noise_points
+
```
Comment (petrasovaa) [inline vector/v.cluster/testsuite/cluster_test.py:108]: Can you merge this into one function, it seems somewhat inefficient to run v.out.ascii twice.

File: vector/v.cluster/testsuite/cluster_test.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        )
+
+        clusterIds_3d = set()
+        for line in ascii_output.splitlines():
+            if line.strip():  # Skip empty lines
+                parts = line.split(",")
+                if len(parts) >= 4:
+                    clusterIds_3d.add(parts[3])
+
+        # print(ascii_output)
```
Comment (petrasovaa) [inline vector/v.cluster/testsuite/cluster_test.py:209]: remove the prints

**Review by petrasovaa — APPROVED** [review]
More tests can be added later.
